packages/typescript-estree/src/useProgramFromProjectService.ts

```diff
@@ -1,26 +1,38 @@
 import path from 'path';
-import type { server } from 'typescript/lib/tsserverlibrary';
 
 import { createProjectProgram } from './create-program/createProjectProgram';
+import type { ProjectServiceSettings } from './create-program/createProjectService';
 import { type ASTAndDefiniteProgram } from './create-program/shared';
 import type { MutableParseSettings } from './parseSettings';
 
 export function useProgramFromProjectService(
-  projectService: server.ProjectService,
+  { allowDefaultProjectForFiles, service }: ProjectServiceSettings,
   parseSettings: Readonly<MutableParseSettings>,
+  hasFullTypeInformation: boolean,
 ): ASTAndDefiniteProgram | undefined {
-  const opened = projectService.openClientFile(
+  const opened = service.openClientFile(
     absolutify(parseSettings.filePath),
     parseSettings.codeFullText,
     /* scriptKind */ undefined,
     parseSettings.tsconfigRootDir,
   );
-  if (!opened.configFileName) {
-    return undefined;
+
+  if (hasFullTypeInformation) {
+    if (opened.configFileName) {
+      if (allowDefaultProjectForFiles.has(parseSettings.filePath)) {
```

Comment:
ditto here - probably worth passing the path through `getCanonicalFileName` to ensure we get a match.